velox/exec/tests/SimpleHttpClientTest.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "folly/json.h"
+#include "velox/common/base/VeloxException.h"
+#include "velox/common/base/tests/GTestUtils.h"
+
+#include <map>
+#include <string>
+#include <thread>
+#include "velox/exec/tests/utils/SimpleHttpClient.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec::test;
+
+class SimpleHttpClientTest : public testing::Test {};
+
+TEST_F(SimpleHttpClientTest, get) {
+  auto content = httpGet("https://facebookincubator.github.io/velox/");
+  ASSERT_FALSE(content.empty());
+}
+
+TEST_F(SimpleHttpClientTest, fail) {
+  VELOX_ASSERT_THROW(
+      httpPost("null", "", std::map<std::string, std::string>{}),
+      "Fail to post to null.");
+  VELOX_ASSERT_THROW(httpGet("null"), "Fail to get from null.");
+}
+
+// Disable this test by default as it relies on the PrestoHiverQueryRunner.
+TEST_F(SimpleHttpClientTest, DISABLED_query_hive_runner) {
+  auto url = "http://localhost:8080/v1/statement?binaryResults=true";
+  auto body = "select * from nation limit 1";
+  std::map<std::string, std::string> header = {
+      {"Content-Type", "text/plain"},
+      {"X-Presto-Catalog", "hive"},
+      {"X-Presto-Schema", "tpch"},
+      {"X-Presto-User", "hive"}};
+  auto resObject = folly::parseJson(httpPost(url, body, header));
+  auto nextUri = resObject["nextUri"].asString();
+  std::string result;
+
+  for (;;) {
+    cpr::Response response = cpr::Get(cpr::Url{nextUri});
+    resObject = folly::parseJson(response.text);
+    if (resObject.count("nextUri") == 0) {
+      break;
+    } else {
+      nextUri = resObject["nextUri"].asString();
+    }
+
+    if (resObject.count("binaryData") != 0 &&
+        !resObject["binaryData"].empty()) {
+      result = folly::toPrettyJson(resObject["binaryData"]);
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
Why sleep?

Reply:
I want to reduce the frequency as it may be in the state of `queued` at the beginning.